WebRoot/js/zimbraMail/prefs/view/ZmFilterRuleDialog.js

```diff
@@ -1550,7 +1550,8 @@ function(inputs, conf, field) {
 		if (!date) {
 			return null;
 		}
-		return String(date.getTime() / 1000);
+		AjxDateUtil._serverDateFormatter = new AjxDateFormat("yyyyMMdd");
+		return AjxDateUtil._serverDateFormatter.format(new Date(date));
```

Comment:
Can we use `AjxDateUtil.getServerDate` ?
https://github.com/Zimbra/zm-ajax/blob/develop/WebRoot/js/ajax/util/AjxDateUtil.js#L598
```suggestion
		return AjxDateUtil.getServerDate(new Date(date));
```

Reply:
ah, the function requires a string in `YYYYMMDDTHHMMSSZ` format...

Reply:
I saw a wrong result with the change.
1. created a filter specifying `Date before Dec/1/2023` in a condition.
2. a created filter has `Aug/17/1970` in a condition. zimbraMailSieveScript has also a wrong date.
```
$ zmprov ga test1 zimbraMailSieveScript
# name test1@dev.zimbra.com
zimbraMailSieveScript: require ["fileinto", "copy", "reject", "tag", "flag", "variables", "log", "enotify", "envelope", "body", "ereject", "reject", "relational", "comparator-i;ascii-numeric"];

# test
if anyof (date :before "19700823") {
    keep;
    stop;
}
```

Reply:
fixed the issue.